### ISSUE TITLE
2 Oracle Versions on LocalPC

### ISSUE THREAD

**Issue by lpotts** [USFS-PNW/Fia-Biosum-Manager#229]
The Oracle .NET Data Provider for Oracle XE 11G conflicts with newer versions of the Oracle client installed on the same computer.

**Comment by jsfried** [comment]
This is a little cryptic. Does that mean that we should not be experiencing this issue?

**Comment by lpotts** [comment]
Jeremy,

BIOSUM was working when both Oracle client and XE were version 11g. I created a GITHUB issue because oracle client 12c and oracle xe 11g broke BIOSUM.  There is an Oracle Data Provider conflict between ODP version used for 11g and ODP version used for 12c.

If you want me to edit the issue description to provide clarity let me know.

As it currently stands, BIOSUM users with both Oracle client version 12c and Oracle XE version 11g installed will not be able to use BIOSUM for calculating wood volumes and biomass.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by jsfried** [comment]
Will uninstalling oracle 12 solve my problem or will I have to also reinstall Oracle 11 (if I can find it)?

Sent via iPhone. Forgive the brevity, typos and lack of nuance.
________________________________

**Comment by lpotts** [comment]
I’ve successfully converted c# code from using Oracle Data Provider (ODP) to using Microsoft Oracle Client.

I tested the FCS Troubleshooter and it successfully returns wood volume and biomass values from FCS oracle with both the simple 1 record test and the batch test.

I have not tested the FIADB plot input process yet.

Jeremy, I would not uninstall oracle 12c or 11g.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lpotts** [comment]
That’s great news! If someone can make an msi I am happy to test the loading (from FIADB and from fvs output)

Sent via iPhone. Forgive the brevity, typos and lack of nuance.
________________________________
From: Potts, Larry D -FS <larry.potts@usda.gov>
Sent: Saturday, January 18, 2020 10:27:58 AM
To: USFS-PNW/Fia-Biosum-Manager <reply@reply.github.com>
Cc: Fried, Jeremy - FS <jeremy.s.fried@usda.gov>; Lesley Bross <lbross@pdx.edu>
Subject: RE: [USFS-PNW/Fia-Biosum-Manager] 2 Oracle Versions on LocalPC (#229)


I’ve successfully converted c# code from using Oracle Data Provider (ODP) to using Microsoft Oracle Client.



I tested the FCS Troubleshooter and it successfully returns wood volume and biomass values from FCS oracle with both the simple 1 record test and the batch test.



I have not tested the FIADB plot input process yet.



Jeremy, I would not uninstall oracle 12c or 11g.



[Forest Service Shield]

Larry Potts
Information Technology Specialist

Forest Service

Resource Monitoring and Assessment

PNW Research Station

Portland Forest Science Lab

Forest Inventory and Analysis

p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>

620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<https://gcc02.safelinks.protection.outlook.com/?url=http%3A%2F%2Fwww.fs.fed.us%2F&data=02%7C01%7C%7C6ec9641abfa7433301ff08d79c4424fb%7Ced5b36e701ee4ebc867ee03cfa0d4697%7C0%7C0%7C637149690522122500&sdata=e6rvBkGP%2FSK8NW17VwEQG4vY1gNPVHKy%2BkkptpCcw5c%3D&reserved=0>
[USDA Logo]<https://gcc02.safelinks.protection.outlook.com/?url=http%3A%2F%2Fusda.gov%2F&data=02%7C01%7C%7C6ec9641abfa7433301ff08d79c4424fb%7Ced5b36e701ee4ebc867ee03cfa0d4697%7C0%7C0%7C637149690522122500&sdata=UoH7lzFM%2FPBwBklT5UYk4SDz6MgtLVp4X5yU9YDr7Bw%3D&reserved=0>[Forest Service Twitter]<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftwitter.com%2Fforestservice&data=02%7C01%7C%7C6ec9641abfa7433301ff08d79c4424fb%7Ced5b36e701ee4ebc867ee03cfa0d4697%7C0%7C0%7C637149690522132497&sdata=uOlLeyxSOvBzU3raNC5BVEGpXRq8X3CFpUVDerr8t3E%3D&reserved=0>[USDA Facebook]<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.facebook.com%2Fpages%2FUS-Forest-Service%2F1431984283714112&data=02%7C01%7C%7C6ec9641abfa7433301ff08d79c4424fb%7Ced5b36e701ee4ebc867ee03cfa0d4697%7C0%7C0%7C637149690522132497&sdata=BwY4HpyNsH%2FFwmTYLMU4ppd4OxdhRS2jGVjr50WUWYw%3D&reserved=0>

Caring for the land and serving people

**Comment by lpotts** [comment]
I did an FIADB plot input test of 5 counties representing over 26000 trees and all live trees had VOLTSGRS values.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lpotts** [comment]
I updated the GITHUB repositories with the MS Oracle Client solution.

These projects are in the GITHUB issue_229 branch:

Fia-Biosum-Manager
Fcs-Class-Library
FIADB-Oracle-Services

They will need to be merged to the master branch.

This project was added to the master branch:
MS-OracleClient


Lesley will need to create the MSI.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lbross** [comment]
@lpotts. I have the updates compiling on my build computer. Some of the settings don't migrate through GitHub so I have a couple of questions regarding the System.Data.OracleClient library in the ado_data_oracle project: 
1. Should CopyLocal be true or false? 
2. Should SpecificVersion be true or false? 
3. The runtime version on my computer is v2.0.50727 and the Version is 2.0.0.0. Are these correct?

I am about to test the .msi on my test computer. After I get answers to these questions, I will post an .msi for you to test on Box. If this tests out, I will merge to the master branch and do a production build for distribution.

**Comment by lpotts** [comment]
SEE BELOW

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lbross** [comment]
Great! The .msi passed all tests on my test computer. It is posted on BiosumBox in the Temp30Days folder. The file name is fia_biosum_setup_vBeta588.msi. Please install and run your tests. If it works for you, I will merge the changes to the master branch(es) and build an .msi for distribution.

**Comment by lbross** [comment]
I just noticed that the GitHub repository is named MS-OracleClient but the assembly and project names are ado_data_oracle. This is confusing as they are the same for all of our other repositories. Should we call it MS-OracleClient or ado_data_oracle? I can make the change either way.

**Comment by lpotts** [comment]
Ado_data_oracle

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by jsfried** [comment]
On a call for about 90 minutes but will try to test while on the call, or else immediately after.

**Comment by jsfried** [comment]
A new error cropped up when running the same plot load from FIADB using this msi:

[cid:image001.png@01D5D04B.EEF5D2F0]

I clicked OK and then it completed. However, VOLTSGRS is null for all records in MASTER.TREE

The biosum calculator button in the troubleshooter tool does work

**Comment by lpotts** [comment]
My recommendation is to not use MS ACCESS.  Non-MS Access applications, like BIOSUM, have to integrate native DAO libraries into the application to create table links which means more application overhead and versioning issues.  I would like to keep it simple and not have to worry about overhead, versioning, and 32bit/64bit issues.

What do you think?

Biosum is a C# .NET product of Visual Studio 2013.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lbross** [comment]
I'm afraid I don't understand the question, or recommendation. We don't have the resources to convert BioSum to a different database platform at this time. Would you like me to build a v5.8.8 .msi with the new ado_data_oracle library or does Jeremy's error require a fix before proceeding?

I think we would expect voltsgrs for year 1 to be null unless the trees were reloaded from scratch because there was an Oracle error when the trees were initially loaded causing voltsgrs to be null in master.tree.

**Comment by jsfried** [comment]
I created a new project and loaded the plumas county data into it from scratch. There should have been a voltsgrs for every live tree.

**Comment by lpotts** [comment]
Ignore the message I sent out, Lesley. It was a conversation I was having with Jason regarding the wood volume compiler he is developing.

Sorry for the confusion.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by jsfried** [comment]
Was very confusing to me too ☺ Happens not infrequently to me when I text…

**Comment by lpotts** [comment]
I installed 5.8.8 and was able to input plot data. All live trees had VOLTSGRS values after the plot input.

About 26,000 trees were processed.

[Forest Service Shield]
Larry Potts
Information Technology Specialist
Forest Service
Resource Monitoring and Assessment
PNW Research Station
Portland Forest Science Lab
Forest Inventory and Analysis
p: 503-808-2079
lpotts@usda.gov<mailto:lpotts@usda.gov>
620 SW Main, Suite 502
Portland, OR 97205
www.fs.fed.us<http://www.fs.fed.us/>
[USDA Logo]<http://usda.gov/>[Forest Service Twitter]<https://twitter.com/forestservice>[USDA Facebook]<https://www.facebook.com/pages/US-Forest-Service/1431984283714112>
Caring for the land and serving people

**Comment by lbross** [comment]
@jsfried Do you want to share your Plumas project with @lpotts so he can look into it before releasing this fix? From what you are saying, you should have voltsgrs, I think.

**Comment by jsfried** [comment]
It was an empty (new project) into which I loaded the state 6 county 63 data from the FIADB that I made from an SQLite db (https://usfs.box.com/s/tdkc35ndgg7s7ce757qqke3g6yv780nw). So there is nothing to share, really, other than this link.

**Comment by jsfried** [comment]
The original problem in this long thread (will teach me to communicate in Github rather than by infinitely long email threads that are hard to follow when transcribed into github) was that ORACLEXE 11g was not compatible with ORACLE 12. Since I am with the current 588, able to load FIADB plots in and FVS output in to BioSum projects (on the one of my computers that still has ORACLEXE working OK) I consider this particular issue (which Larry fixed by switching from ODP to a Microsoft alternative) to be fixed.
